src/NServiceBus.TransactionalSession.AcceptanceTests.AzureTable/When_using_transactional_session.cs

```diff
@@ -62,6 +62,51 @@ await Scenario.Define<Context>()
             Assert.AreEqual(tableEntity.Properties["Data"].StringValue, "MyCustomData");
         }
 
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task Should_send_messages_and_store_document_in_sql_session_on_transactional_session_commit(bool outboxEnabled)
+        {
+            var entityRowId = Guid.NewGuid().ToString();
+
+            await Scenario.Define<Context>()
+                .WithEndpoint<AnEndpoint>(s => s.When(async (_, ctx) =>
+                {
+                    using var scope = ctx.ServiceProvider.CreateScope();
+                    using var transactionalSession = scope.ServiceProvider.GetRequiredService<ITransactionalSession>();
+                    await transactionalSession.OpenAzureTableSession(PartitionKeyValue, AzureTableSetup.TableName);
+
+                    var sendOptions = new SendOptions();
+                    sendOptions.SetHeader(PartitionKeyHeaderName, PartitionKeyValue);
+                    sendOptions.RouteToThisEndpoint();
+
+                    await transactionalSession.Send(new SampleMessage(), sendOptions, CancellationToken.None);
+
+                    var storageSession = scope.ServiceProvider.GetRequiredService<IAzureTableStorageSession>();
```

Comment:
Could we use test cases for this? I assume that this is the only line that is different between taking session from `ITransactionalSession.SynchronizedSession` vs from DI

Reply:
We would then have to have

- Outbox True / InjectedSession
- Outbox True / TransactionalSession Session
- Outbox False  / InjectedSession
- Outbox False  / TransactionalSession Session

not sure that is worth it

Reply:
I think it's not worth yet to introduce additional complexity to the test infrastructure to avoid this little bit of duplication. Once we'd like to do that, this can be easily done via a separate PR.